app/src/main/java/com/ataulm/whatsnext/search/SearchViewModel.kt

```diff
@@ -22,35 +14,16 @@ internal class SearchViewModel(private val service: WhatsNextService) : ViewMode
     private val _navigationEvents = MutableLiveData<Event<FilmSummary>>()
     val navigationEvents: LiveData<Event<FilmSummary>> = _navigationEvents
 
-    private var disposable: Disposable? = null
-
     fun onSearch(searchTerm: String) {
-        disposable?.dispose()
-        disposable = service.search(searchTerm)
-                .subscribeOn(Schedulers.io())
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribeWith(
-                        object : ErrorTrackingDisposableObserver<List<FilmSummary>>() {
-                            override fun onNext(filmSummaries: List<FilmSummary>) {
-                                Log.d("!!!", "onNext $filmSummaries")
-                                _films.value = filmSummaries
-                            }
-
-                            override fun onComplete() {
-                                Log.d("!!!", "onComplete")
-                            }
-                        }
-                )
+        viewModelScope.launch {
```

Comment:
As [in this post by Manuel](https://medium.com/androiddevelopers/easy-coroutines-in-android-viewmodelscope-25bffb605471), `viewModelScope` is an extension property whose associated job will be cancelled `ViewModel.onCleared()` so we don't have to do it ourselves.